packages/core/src/jsonWeb/verifyJsonWebToken.ts

```diff
@@ -0,0 +1,37 @@
+import type { VerifyCallback } from '../utils'
+import { getUsedJsonWebKey } from './getUsedJsonWebKey'
+import type { JsonWebKeySet } from './jsonWebKeySet'
+import { parseJsonWebToken } from './parseJsonWebToken'
+
+type VerifyJsonWebTokenOptions = {
+  verifyJwtCallback: VerifyCallback
+  jwks?: JsonWebKeySet
+  jwt: string
+}
+
+export const verifyJsonWebToken = async ({ jwt, jwks, verifyJwtCallback }: VerifyJsonWebTokenOptions) => {
+  const { header, signature, claims, signableInput } = parseJsonWebToken(jwt)
+
+  const jsonWebKeySetClaims = jwks ? { jwks } : claims
+
+  const jwk = getUsedJsonWebKey(header, jsonWebKeySetClaims)
+
+  try {
+    const isValid = await verifyJwtCallback({
+      signature,
+      jwk,
+      data: signableInput,
+    })
+
+    // TODO: better error message
+    if (!isValid) {
+      throw new Error('Signature in the JWT is invalid')
+    }
+  } catch (e: unknown) {
+    if (typeof e === 'string') {
```

Comment:
Should probably also handle error not being instanceof error, so we can assume when calling this library that always an error will be thrown, even if the callback didn't throw an error instance